tilelang/backend/target.py

```
@@ -0,0 +1,307 @@
+    raise TypeError(f"Unsupported target type: {type(target)!r}")
+
+
+def _detect_auto_target() -> TargetInput:
+    detectors = [kind for kind in _TARGET_KINDS.values() if kind.detect is not None]
+    detectors.sort(key=lambda kind: kind.priority, reverse=True)
+    errors: list[str] = []
+    for kind in detectors:
+        assert kind.detect is not None
+        try:
+            detected = kind.detect()
+        except Exception as err:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)